eoxserver/services/ows/wps/processes/get_dem_processing.py

```diff
@@ -0,0 +1,235 @@
+            title="output data",
+            abstract="Binary/geojson complex data output.",
+            formats=(
+                FormatBinaryRaw('image/png'),
+                FormatBinaryBase64('image/png'),
+                FormatBinaryRaw('image/jpeg'),
+                FormatBinaryBase64('image/jpeg'),
+                FormatBinaryRaw('image/tiff'),
+                FormatBinaryBase64('image/tiff'),
+                FormatJSON()
+            )
+        ),
+    }
+
+    @staticmethod
+    def execute(coverage, identifier, bbox, result, z_factor, interval, scale, azimuth, altitude):
+        """ The main execution function for the process.
+        """
+
+        values = list(map(float, bbox.split(",")))
+        if len(values) != 4:
+            raise ValueError("Invalid number of coordinates in 'bbox'.")
+
+        data_format = "raster"
+        # output format selection
+        if result['mime_type'] == "image/png":
+            extension = ".png"
+            driver = gdal.GetDriverByName("PNG")
+        elif result['mime_type'] == "image/jpeg":
+            extension = ".jpg"
+            driver = gdal.GetDriverByName("JPEG")
+        elif result['mime_type'] == "image/tiff":
+            extension = ".tif"
+            driver = gdal.GetDriverByName("GTiff")
+        else:
+            extension = ".geojson"
+            data_format = "vector"
+            driver = gdal.GetDriverByName("GeoJSON")
+
+        # get the dataset series matching the requested ID
+        try:
+            model = models.Coverage.objects.get(
+                identifier=coverage)
+
+        except models.Coverage.DoesNotExist:
+            raise InvalidInputValueError(
+                "coverage", "Invalid coverage name '%s'!" % coverage
+            )
+        try:
+
+            data_items = model.arraydata_items.all()
+
+        except model.arraydata_items.all().length > 1:
+            raise InvalidInputValueError(
+                "coverage", "coverage '%s' has more than one imagery, the profile process handles single images!" % coverage
+            )
+        data_item = data_items[0]
+
+        original_ds = gdal_open(data_item, False)
+
+        # check if the provided box is compatible with the coverage
+        geoTransform = original_ds.GetGeoTransform()
+        minx = geoTransform[0]
+        maxy = geoTransform[3]
+        maxx = minx + geoTransform[1] * original_ds.RasterXSize
+        miny = maxy + geoTransform[5] * original_ds.RasterYSize
+        coverage_bbox = Polygon.from_bbox((minx, miny, maxx, maxy))
+        request_bbox = Polygon.from_bbox(values)
+
+        if coverage_bbox.contains(request_bbox):
+            values_bbox = coverage_bbox.intersection(request_bbox)
+            if values_bbox.area > 0:
+                values = list(values_bbox.extent)
+            else:
+                logger.error('The provided bbox is not inside or intersecting with the coverage')
+
+        output_filename = "/opt/data/result" + extension
```

Comment:
It may be quite slow to write the results to the disc, only to read them again afterwards. Maybe we could use `vsimem` here as well?